[INTPLAT-457] DDSaaS: Postmark: Integration v1.0.0 (#18898)
DataDog/integrations-core: postmark/README.md

# Postmark

## Overview

[Postmark][1] is an email delivery service for sending and tracking application emails. It offers a scalable alternative to traditional SMTP and provides essential metrics like sent, opened, bounced, and spam complaint rates. This enables efficient email management and monitoring.

To gain insights into Postmark's broadcast and transactional message streams, integrate Postmark with Datadog using [webhooks][2].

## Setup

Follow the instructions below to configure this integration for your Postmark account.

### Configuration

#### Enable open and link tracking in server settings for the message streams
Follow these steps to enable both features:

> 1. Log in to your [Postmark account][3]. This redirects you to the [servers page][4].
> 2. Select the desired server. This redirects you to the **Message Streams page**.
> 3. In the navigation panel, click the **Settings** tab.
> 4. In the Tracking section, enable both **Open tracking** and **Link tracking**.

Reviewer: ```suggestion
1. Log into your [Postmark account][3]. This redirects you to the [servers][4] page.
2. Select the desired server. This redirects you to the **Message Streams** page.
3. In the navigation panel, click the **Settings** tab.
4. In the **Tracking** section, enable both **Open tracking** and **Link tracking**.
```

#### Webhook configuration steps

> Configure the Datadog endpoint to forward below activity logs to Datadog. [See [Postmark Webhook Overview][2] for more details]
> - **Broadcast** message streams
> - **Transactional** message streams
> 
> Perform below steps to configure webhook for both message streams mentioned above.
> 
> 1. Select an existing API key or create a new one by clicking one of the buttons below:<!-- UI Component to be added by DataDog team -->
> 2. Log in to your [Postmark account][3]. This redirects you to the [servers page][4].
> 3. Select the desired server. This redirects you to the **Message Streams page**.

Reviewer: ```suggestion
Configure the Datadog endpoint to forward the following activity logs to Datadog: 
- **Broadcast** message streams
- **Transactional** message streams

See [Postmark Webhook Overview][2] for more details.

Perform the following steps to configure the webhook for both of the message streams mentioned above.

1. Select an existing API key or create a new one by clicking one of the buttons below: <!-- UI Component to be added by DataDog team -->
2. Log into your [Postmark account][3]. This redirects you to the [servers][4] page.
3. Select the desired server. This redirects you to the **Message Streams** page.
```

> 4. On the message streams page, select an existing broadcast / transactional message stream or create a new one. 

Reviewer: ```suggestion
4. On the message streams page, select an existing broadcast or transactional message stream or create a new one. 
```

## Setup

To configure the integration for Postmark broadcast and transactional message streams activity logs through a webhook, follow these steps.

### Configuration
#### Enable open and link tracking in server settings for the message streams
Follow these steps to enable both features:

1. Log in to your [Postmark account][3]. This redirects you to the [servers page][4].
2. Select the desired server. This redirects you to the **Message Streams page**.
3. In the navigation panel, click the **Settings** tab.
4. In the Tracking section, enable both **Open tracking** and **Link tracking**.

#### Webhook configuration steps for broadcast message streams
Configure the Datadog endpoint to forward Postmark broadcast message streams activity logs to Datadog.

1. Select an existing API key or create a new one by clicking one of the buttons below:<!-- UI Component to be added by DataDog team -->
2. Log in to your [Postmark account][3]. This redirects you to the [servers page][4].
3. Select the desired server. This redirects you to the **Message Streams page**.

> 4. 4. On the message streams page, either select an existing broadcast message stream or create a new one. To create a new stream, click the **Create Message Stream** button, enter a stream name, and select **Broadcasts** as the message type.

Reviewer 2: ```suggestion
4. On the message streams page, either select an existing broadcast message stream or create a new one. To create a new stream, click the **Create Message Stream** button, enter a stream name, and select **Broadcasts** as the message type.
```

# Postmark

## Overview

[Postmark][1] is an email delivery service for sending and tracking application emails. It offers a scalable alternative to traditional SMTP and provides essential metrics like sent, opened, bounced, and spam complaint rates. This enables efficient email management and monitoring.

To gain insights into Postmark's broadcast and transactional message streams, integrate Postmark with Datadog using [webhooks][2].

## Setup

Follow the instructions below to configure this integration for your Postmark account.

### Configuration

#### Enable open and link tracking in server settings for the message streams
Follow these steps to enable both features:

1. Log in to your [Postmark account][3]. This redirects you to the [servers page][4].
2. Select the desired server. This redirects you to the **Message Streams page**.
3. In the navigation panel, click the **Settings** tab.
4. In the Tracking section, enable both **Open tracking** and **Link tracking**.

> #### Webhook configuration steps for broadcast message streams

Reviewer 3: can we combine the instructions for message streams and transcation message streams into the same steps and just differentiate where the instructions are different? Looking at the config instructions, both webhook config steps are basically identical.

Author: Done
